plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -420,14 +424,14 @@ private Slice writeDeletionVector(
 
         try {
             DataFileInfo newFileInfo = new DataFileInfo(
-                    sourceRelativePath,
+                    sourceReferencePath,
                     length,
                     lastModified.toEpochMilli(),
                     DATA,
                     deletion.partitionValues,
                     readStatistics(parquetMetadata, dataColumns, rowCount),
                     Optional.of(deletionVectorEntry));
-            DeltaLakeMergeResult result = new DeltaLakeMergeResult(deletion.partitionValues, Optional.of(sourceRelativePath), Optional.ofNullable(oldDeletionVector), Optional.of(newFileInfo));
+            DeltaLakeMergeResult result = new DeltaLakeMergeResult(deletion.partitionValues, Optional.of(sourceReferencePath), Optional.ofNullable(oldDeletionVector), Optional.of(newFileInfo));
```

Comment:
Thank you for adding deletionVector test case, what I tried to achieve with this request is to exercise this line in test. With only one row Trino does not write any deletion vector https://github.com/trinodb/trino/blob/c76bcf1f3130b37e46598fae48160e968124e17d/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java#L393-L395
The best would be to create another source table with more rows, which also have deletionVector enabled.  Add also some updates so we can test referring to deletion vector files from source table.

Reply:
Thanks for the elaborating

Reply:
updated, PTAL